src/internal.c

```diff
@@ -8435,6 +8435,13 @@ void wolfSSL_ResourceFree(WOLFSSL* ssl)
     }
     FreeSuites(ssl);
     FreeHandshakeHashes(ssl);
+#ifdef HAVE_ECH
+    /* try to free the ech hashes in case we errored out */
+    ssl->hsHashes = ssl->hsHashesEch;
```

Comment:
Assigning to be freed seems silly.
Make a function that takes a HsHashes* and frees it.
Either name it FreeHanshakeHashes and change callers or have FreeHanshakeHashes call the new function.